tsconfig.json

```
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "forceConsistentCasingInFileNames": true,
+    "isolatedModules": true,
+    "jsx": "react",
+    "lib": ["es6", "dom"],
+    "module": "commonjs",
+    "noEmit": true,
+    "strict": true,
+    "target": "es5"
+  },
+  "include": ["packages", "scripts", "site", "test", "playgrounds"],
+  "exclude": ["node_modules"]
```

I've found myself doing this in my own integration branch to get TypeScript to not complain about all the `$ExpectError` dts tests:

```suggestion
  "exclude": ["node_modules", "packages/*/types/asdf*"]
```

---

I guess, more broadly, what is the strategy for `.d.ts` generation? Should a build step like `build:types` be added that runs `tsc --declaration` into `types/` folders?

---

> I guess, more broadly, what is the strategy for .d.ts generation? Should a build step like build:types be added that runs tsc --declaration into types/ folders?

This one is covered by Preconstruct which we are using to build all packages. You can check out example output containing TS declarations here:
https://unpkg.com/browse/@changesets/cli@2.14.0/dist/